pyvista/utilities/reader.py

```diff
@@ -1478,29 +1483,97 @@ def set_active_time_point(self, time_point):  # noqa: D102
         self.set_active_time_value(self.time_values[time_point])
 
 
+class DICOMReader(BaseReader):
+    """DICOM Reader for reading ``dcm`` files.
+
+    This reader reads a single file or a path containing a several ``dcm``
+    files (DICOM stack).
+
+    Parameters
+    ----------
+    path : str
+        Path to the single DICOM (``dcm``) file to be opened or the directory
+        containing a stack of DICOM files.
+
+    Raises
+    ------
+    FileNotFoundError
+        If ``path`` is not a DICOM file (``dcm``) or a directory containing
+        DICOM files.
+
+    Examples
+    --------
+    >>> import pyvista
+    >>> from pyvista import examples
+    >>> path = examples.download_dicom_stack(load=False)
+    >>> reader = pyvista.DICOMReader(path)
+    >>> dataset = reader.read()
+    >>> dataset.plot(volume=True, zoom=3, show_scalar_bar=False)
+
+    """
+
+    _class_reader = _vtk.vtkDICOMImageReader
+
+    def __init__(self, path: str):
+        """Initialize reader from DICOM file (*.dcm) or folder of DICOM files."""
+        self._reader = self._class_reader()
+        self._progress_bar = False
+        self._progress_msg = None
+        self.directory = None
+
+        if os.path.isfile(path):
+            self.filename = path
+            self._set_filename(path)
+        elif os.path.isdir(path):
+            self.directory = path
+
+            if not os.listdir(path):
+                raise FileNotFoundError(f"No DICOM (*.dcm) files found in directory\n\n{path}")
+
+            file_types = set(pathlib.Path(file_).suffix for file_ in os.listdir(path))
+            file_types.discard('')
+
+            if file_types != {".dcm"}:
```

Comment:
We aren't doing these checks for a single file.  But I'm not sure that we should be policing whether the input files have a suffix that matches, unless there is a reason for this particular reader.  For example, ANSYS Fluent can output Ensight files which are nice to read into vtk, but it outputs with nonstandard suffixes.  These sorts of checks would require the user to rename the files or subclass the reader.  Both not attractive options.  If we want to have a check for suffix, I would prefer a warning, although if there is no file or folder at all than the error makes sense.

For a future PR, it would be good to move this logic into the base class to keep the behavior consistent across Readers for these types of checks.

Reply:
Opened https://github.com/pyvista/pyvista/issues/2471

Reply:
I don't love this piece of logic. If you strip the extension off the files in a DICOM directory (or change them to anything else, or even give the files all different filenames), `vtkDICOMImageReader` will still read in the image slices and assemble the DICOM correctly.

It should be up to the user to have a proper DICOM directory with only DICOM files in it in the correct format as per the requirements of the underlying `vtk` image reader and there shouldn't be extra constraints added in my opinion.

Reply:
I am completely fine to remove file extension checks from the code. VTK outputs a warning message if it cannot parse a particular file; I wasn't sure if we wanted to handle warnings differently in `pyvista` or just let VTK messages pass through.

I agree with both @MatthewFlamm and @njneeteson.